be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -154,6 +164,11 @@ Status HdfsTextScanner::parse_csv(int chunk_size, ChunkPtr* chunk) {
     }
 
     csv::Converter::Options options;
+    // Use to custom Hive array format
+    options.array_format_type = csv::ArrayFormatType::kHive;
+    options.array_hive_collection_delimiter = _collection_delimiter.front();
```

Comment:
Either change the type of `_collection_delimiter` to char or check if the size of `_collection_delimiter` equals one.